src/awkward/operations/structure.py

```diff
@@ -1460,6 +1460,60 @@ def apply(layout):
         return out
 
 
+def unflatten(array, counts, highlevel=True):
```

Comment:
We can leave the problem of how to add an `axis` argument for later, but since it should go before the `highlevel=True` (that's usually last, with the exception of very low-level arguments), it's a good idea to add it now, to ensure backward compatibility:

```python
def unflatten(array, counts, axis=0, highlevel=True):
```

(Just in case someone sets `highlevel` by position, we don't want their code to break when `axis` is added later.)

Then in the body of the function, raise a NotImplementedError if `axis` is not `0`.

Reply:
No axis! See https://github.com/scikit-hep/awkward-1.0/pull/583#issuecomment-741939466.